src/App.tsx

```diff
@@ -7,218 +7,91 @@ import { Filter } from "./panelSections/Filter";
-    channelIn.addListener("midimessage", (e) => {
-      const [_, parameter, value] = e.data;
-      const knobValue = Math.round((value / 123) * 1027);
-      const threePoleValue = (value / 123) * 2;
-      const fourPoleValue = (value / 123) * 3;
-      switch (parameter) {
-        // GLOBAL
-        case PARAMS.DRIVE:
-          setDrive(knobValue);
-          break;
-        // VCO1
-        case PARAMS.VCO1_WAVE:
-          setVco1Wave(threePoleValue);
-          break;
-        case PARAMS.VCO1_SHAPE:
-          setVco1Shape(knobValue);
-          break;
-        // VCO2
-        case PARAMS.VCO2_OCTAVE:
-          setVco2Octave(fourPoleValue);
-          break;
-        case PARAMS.VCO2_PITCH:
-          setVco2Pitch(knobValue);
-          break;
-        case PARAMS.VCO2_WAVE:
-          setVco2Wave(threePoleValue);
-          break;
-        case PARAMS.VCO2_DUTY:
-          setVco2Duty(threePoleValue);
-          break;
-        case PARAMS.VCO2_SHAPE:
-          setVco2Shape(knobValue);
-          break;
-        // MIXER
-
-        case PARAMS.VCO1_LEVEL:
-          setVco1Level(knobValue);
-          break;
-        case PARAMS.VCO2_LEVEL:
-          setVco2Level(knobValue);
-          break;
 
-        // FILTER
-        case PARAMS.CUTOFF:
-          setCutoff(knobValue);
-          break;
-        case PARAMS.RESONANCE:
-          setResonance(knobValue);
-          break;
+  const connectMidi = async () => {
+    return monologueController.connect();
+  };
 
-        // EG
-        case PARAMS.ENV_TYPE:
-          setEnvType(threePoleValue);
-          break;
-        case PARAMS.ENV_ATTACK:
-          setEnvAttack(knobValue);
-          break;
-        case PARAMS.ENV_DECAY:
-          setEnvDecay(knobValue);
-          break;
-        case PARAMS.ENV_INTENSITY:
-          setEnvIntensity(knobValue);
-          break;
-        case PARAMS.ENV_TARGET:
-          setEnvTarget(threePoleValue);
-          break;
+  const [patchName, setPatchName] = useState(() => korgProgramDump.patchName);
+  const [paramState, setParamState] = useState(() => initialiseParamState(korgProgramDump));
 
-        // LFO
-        case PARAMS.LFO_WAVE:
-          setLfoWave(threePoleValue);
-          break;
-        case PARAMS.LFO_MODE:
-          setLfoMode(threePoleValue);
-          break;
-        case PARAMS.LFO_RATE:
-          setLfoRate(knobValue);
-          break;
-        case PARAMS.LFO_INTENSITY:
-          setLfoIntensity(knobValue);
-          break;
-        case PARAMS.LFO_TARGET:
-          setLfoTarget(threePoleValue);
-          break;
+  const appliedParamState = getMergedParamState(paramState, setParamState, monologueController);
+  const setParamViaCallback = getMergedParamStateForCallback(paramState, setParamState, monologueController);
 
-        default:
-          console.log(e.data);
-          break;
-      }
-    });
-  };
+  monologueController.onParameterChange(Parameters.DRIVE, appliedParamState);
```

Comment:
This code here is a little sensitive. It was previously in an effect that only ran on mount, but because I'm using partial application to create the `appliedParamState` function with all it's dependencies, the `paramState` was effectively getting snapshotted to it's value on mount, and wasn't getting updated on each render.